tests/background/sidebar-injector-test.js

```diff
@@ -76,10 +102,31 @@ describe('SidebarInjector', function () {
         getURL: sinon.spy(path => EXTENSION_BASE_URL + path),
       },
 
+      permissions: {
+        request: sinon.stub().callsFake(async request => {
+          const allowed = request.permissions.every(perm =>
+            allowedPermissions.includes(perm)
+          );
+          if (allowed) {
+            request.permissions.forEach(perm => permissions.add(perm));
+          }
+          return allowed;
+        }),
+      },
+
```

Comment:
The mocking for things in the Chrome API in this test module is relatively involved; I don't know if there's any alternative...

Reply:
I guess the main thing that we could do is to centralize the mocks if this logic gets repeated across multiple tests. So far only this one class relies on this more complex faking.